experimental/oak_baremetal_channel/SPEC.MD

```diff
@@ -0,0 +1,290 @@
+# Oak Baremetal Communication Protocol Specification
+
+This document serves as a specification of the protocol used for communication
+between the Untrusted Loader and Trusted Runtime. The Trusted Runtime should be
+able to communicate with any implementation of the Untrusted Loader that
+implements this specification.
+
+In the specification below the Trusted Runtime will henceforth be referred to
+the as the "service". The Untrusted Loader will be referred to the as the
+"client".
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://www.rfc-editor.org/rfc/rfc2119).
+
+## Layer Defintions
+
+### Method Layer
+
+Methods are exposed by the service. Each method has clearly defined parameters
+and return types. The Oak IDL is used to define which methods are available,
+their id, parameter and return types, as well the protcol used to encode
+parameters and return values to bytes.
+
+#### Invocation
+
+An invocation of a Method. Each invocation consists out of a request followed by
+a response.
+
+##### Initiating an invocation
+
+The client MAY initiate an invocation by sending a request.
+
+The service MUST NOT send requests.
+
+##### Handling an invocation
```

Comment:
Which layer does [`schema.fbs`](https://github.com/project-oak/oak/blob/main/experimental/oak_baremetal_channel/schema.fbs) belong to?

Reply:
Method layer :)